cloud/compute-engine/cloud-init/variables.tf

```diff
@@ -0,0 +1,120 @@
+variable "name" {
+  type        = string
+  description = "Name of the instance and managed instance group"
+}
+
+variable "description" {
+  type        = string
+  description = "Description of the instance template"
+  default     = "Compute Engine instance with cloud-init support"
+}
+
+variable "zone" {
+  type        = string
+  description = "GCP zone for the instance group (e.g., us-central1-a)"
+}
+
+variable "machine_type" {
+  type        = string
+  description = "Machine type for the instance (e.g., e2-micro, e2-small, e2-medium)"
+  default     = "e2-micro"
+}
+
+variable "instance_count" {
+  type        = number
+  description = "Number of instances to create in the managed instance group"
+  default     = 1
+}
+
+variable "boot_disk" {
+  type = object({
+    image   = string
+    size_gb = number
+    type    = string
+  })
+  description = "Boot disk configuration including image, size, and type"
+  default = {
+    image   = "ubuntu-os-cloud/ubuntu-2204-lts"
+    size_gb = 10
+    type    = "pd-standard"
+  }
+}
+
+variable "network" {
+  type = object({
+    vpc         = string
+    subnet      = string
+    external_ip = bool
+  })
+  description = "Network configuration including VPC, subnet, and external IP settings"
+}
+
+variable "service_account_email" {
```

Comment:
this module should create and manage the service account for the instances so the service account can't be accidentally deleted independently of the instances